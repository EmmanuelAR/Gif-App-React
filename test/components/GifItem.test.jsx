import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test del componente GifItem", () => {
  const titulo = "CheeseBurger";
  const url = "https://cheeseburger/cheseeburger.jpg";

  test("Debe de hacer match con el SnapShot", () => {
    const { container } = render(<GifItem title={titulo} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar imagen con el url y titulo indicado", () => {
    render(<GifItem title={titulo} url={url} />);
    screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(titulo);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(titulo);
  });
});
