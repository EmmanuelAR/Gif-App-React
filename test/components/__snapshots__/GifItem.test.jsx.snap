// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente GifItem Debe de hacer match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="CheeseBurger"
      src="https://cheeseburger/cheseeburger.jpg"
    />
    <p>
      CheeseBurger
    </p>
  </div>
</div>
`;
